app/routes.py

```diff
@@ -1 +1,302 @@
-from flask import Blueprint
\ No newline at end of file
+from flask import Blueprint, jsonify, request, make_response, abort
+from app import db
+from app.models.task import Task
+from app.models.goal import Goal
+from datetime import date
+
+
+
+
+
+
+tasks_bp = Blueprint("tasks", __name__, url_prefix="/tasks")
+goals_bp = Blueprint("goals", __name__, url_prefix="/goals")
+
+
+
+                            #### HELPER FUNCTIONS ####
+
+def return_task_or_abort(task_id):
+    try:
+        verified_id = int(task_id)
+    except ValueError:
+        abort(make_response({"details":"Invalid ID: id must be an integer"}, 400))
+
+    task = Task.query.get(verified_id)
+
+    if not task:
+        abort(make_response({"details":"Invalid ID: id does not exist"}, 404))
+
+    return task
+
+
+
+def return_goal_or_abort(goal_id):
+    try:
+        verified_id = int(goal_id)
+    except ValueError:
+        abort(make_response({"details":"Invalid ID: id must be an integer"}, 400))
+
+    goal = Goal.query.get(verified_id)
+
+    if not goal:
+        abort(make_response({"details":"Invalid ID: id does not exist"}, 404))
+
+    return goal
+
+
+def boolean_completeness(completed_at):
```

Comment:
This is a great helper function but would make sense as a method in the Task class.